### PR TITLE
Add setup.py

### PR DESCRIPTION
**Pull request by kotfic**
@cjh1 PTAL

Before testing:
1. move ```config.json``` into ```cumulus/conf/config.json```  (will fix this in HPCCloud-deploy once this looks good)
2. ```sudo service girder stop```
3. ```sudo -iu hpccloud```
4. ```cd /opt/hpccloud/cumulus```
5. ```python setup.py install --user```
4. ```cd /opt/hpccloud/girder && girder -m python```

### CODE REVIEW

**Comment by kotfic** [comment]
Oh,  also I am not as familiar with cmake,  is there something that needs to happen to make sure we're installing rather than setting PYTHONPATH?

**Comment by cjh1** [comment]
Looks good, a couple of things:
- The tests/ directory seems to be installed, not sure we want the unit test installed?
- We will need to update .travis.yml, to use the install rather than setting the PYTHONPATH  ( a similar thing will need to be done for HPCCloud-deploy ).  The Travis tests will then fail I am guessing. 
- The following template will need to be install https://github.com/Kitware/cumulus/tree/master/scripts ( we don't need codes, as these are deployed to girder ) and then the following code will need to be fixed up to find them in the right place https://github.com/Kitware/cumulus/blob/master/cumulus/starcluster/tasks/job.py#L164-L175

File: setup.py [hunk 1]
```diff
@@ -0,0 +1,15 @@
+from setuptools import setup, find_packages
+
+setup(
+    name="cumulus",
+    version="0.1.0",
+    description="Girder API endpoints for interacting with cloud providers.",
```
Comment (cjh1) [inline setup.py]: This should probably be "A RESTful API for the creation and management of HPC clusters"

**Comment by cjh1** [comment]
LGTM

**Pull request merged**